### DISCUSSION TITLE
Why does ipsec SA rebuild frequently

### DISCUSSION THREAD

**Discussion by clyuz** [strongswan/strongswan#1867] [Q&A]
My current environment has two VMs, each with two virtual network dev corresponding to two network planes; I have established two IPSec encryption tunnels between two VMs, and there is no problem using either tunnel to ping each other between the two VMs, but only one IPSec SA exists simultaneously; When I use the ping command on both planes at the same time, the SA displayed in the "ip xfrm state" will switch back and forth for only 1 second, resulting in network congestion on both planes.
I am using Ovs to create and manage IPSec.
## Environment:
OS: Suse12.5
strongswan: 5.8.2
ovs: 2.14.2
### ipsec.conf
\# Generated by ovs-monitor-ipsec...do not modify by hand!

config setup
    uniqueids=yes

conn %default
    keyingtries=%forever
    type=transport
    keyexchange=ikev2
    auto=route
    ike=aes256gcm16-sha256-modp2048
    esp=aes256gcm16-modp2048

ca ca_auth
    cacert=/etc/keys/cacert.pem

conn ipsec_a2bc7f61-1
    left=0.0.0.0
    right=xxx.xxx.0.167
    leftid=bunker_ipsec
    rightid=bunker_ipsec
    leftcert=/etc/keys/bunker_ipsec-cert.pem
    leftprotoport=gre
    rightprotoport=gre

conn ipsec_9f44b2f4-1
    left=0.0.0.0
    right=xxx.xxx.99.42
    leftid=bunker_ipsec
    rightid=bunker_ipsec
    leftcert=/etc/keys/bunker_ipsec-cert.pem
    leftprotoport=gre
    rightprotoport=gre
### ipsec statusall
Status of IKE charon daemon (strongSwan 5.8.2, Linux 4.12.14-122.144-default, x86_64):
  uptime: 43 hours, since Aug 26 14:32:15 2023
  malloc: sbrk 5218304, mmap 0, used 3783776, free 1434528
  worker threads: 10 of 16 idle, 6/0/0/0 working, job queue: 0/0/0/0, scheduled: 112
  loaded plugins: charon-systemd ldap pkcs11 aes des blowfish rc2 sha2 sha1 md4 md5 mgf1 random nonce x509 revocation constraints pubkey pkcs1 pkcs7 pkcs8 pkcs12 pgp dnskey sshkey pem openssl af-alg fips-prf gmp curve25519 agent xcbc cmac hmac ctr ccm gcm drbg curl soup attr kernel-netlink resolve socket-default farp stroke vici smp updown eap-identity eap-sim eap-sim-pcsc eap-aka eap-aka-3gpp2 eap-simaka-pseudonym eap-simaka-reauth eap-md5 eap-gtc eap-mschapv2 eap-dynamic eap-radius eap-tls eap-ttls eap-peap eap-tnc xauth-generic xauth-eap xauth-pam tnc-imc tnc-imv tnc-tnccs tnccs-20 tnccs-11 tnccs-dynamic dhcp certexpire led duplicheck radattr addrblock unity counters
Listening IP addresses:
  xxx.xxx.0.142
  xxx.xxx.99.160
  172.77.0.1
  172.66.0.1
Connections:
ipsec_a2bc7f61-1:  0.0.0.0...xxx.xxx.0.167  IKEv2
ipsec_a2bc7f61-1:   local:  [bunker_ipsec] uses public key authentication
ipsec_a2bc7f61-1:    cert:  "C=US, ST=CA, O=Open vSwitch, OU=Open vSwitch certifier, CN=bunker_ipsec"
ipsec_a2bc7f61-1:   remote: [bunker_ipsec] uses public key authentication
ipsec_a2bc7f61-1:   child:  dynamic[gre] === dynamic[gre] TRANSPORT
ipsec_9f44b2f4-1:  0.0.0.0...xxx.xxx.99.42  IKEv2
ipsec_9f44b2f4-1:   local:  [bunker_ipsec] uses public key authentication
ipsec_9f44b2f4-1:    cert:  "C=US, ST=CA, O=Open vSwitch, OU=Open vSwitch certifier, CN=bunker_ipsec"
ipsec_9f44b2f4-1:   remote: [bunker_ipsec] uses public key authentication
ipsec_9f44b2f4-1:   child:  dynamic[gre] === dynamic[gre] TRANSPORT
Routed Connections:
ipsec_9f44b2f4-1{6}:  ROUTED, TRANSPORT, reqid 4
ipsec_9f44b2f4-1{6}:   xxx.xxx.99.160/32[gre] === xxx.xxx.99.42/32[gre]
ipsec_a2bc7f61-1{5}:  ROUTED, TRANSPORT, reqid 3
ipsec_a2bc7f61-1{5}:   xxx.xxx.0.142/32[gre] === xxx.xxx.0.167/32[gre]
Security Associations (1 up, 1 connecting):
   (unnamed)[2963]: CONNECTING, xxx.xxx.0.142[%any]...xxx.xxx.0.167[%any]
   (unnamed)[2963]: IKEv2 SPIs: 900d5e33d465b4cc_i 58b9403a85431a6a_r*
   (unnamed)[2963]: IKE proposal: AES_GCM_16_256/PRF_HMAC_SHA2_256/MODP_2048
   (unnamed)[2963]: Tasks passive: IKE_CERT_PRE IKE_AUTH IKE_CERT_POST IKE_CONFIG CHILD_CREATE IKE_AUTH_LIFETIME IKE_MOBIKE
ipsec_9f44b2f4-1[2939]: ESTABLISHED 21 minutes ago, xxx.xxx.99.160[bunker_ipsec]...xxx.xxx.99.42[bunker_ipsec]
ipsec_9f44b2f4-1[2939]: IKEv2 SPIs: 360f56b7865f874a_i* 8b4b4882937ab8e2_r, public key reauthentication in 2 hours
ipsec_9f44b2f4-1[2939]: IKE proposal: AES_GCM_16_256/PRF_HMAC_SHA2_256/MODP_2048
ipsec_9f44b2f4-1{112}:  INSTALLED, TRANSPORT, reqid 4, ESP SPIs: cc3b029c_i c1254456_o
ipsec_9f44b2f4-1{112}:  AES_GCM_16_256, 132804 bytes_i (1302 pkts, 1s ago), 132804 bytes_o (1302 pkts, 266s ago), rekeying in 21 minutes
ipsec_9f44b2f4-1{112}:   xxx.xxx.99.160/32[gre] === xxx.xxx.99.42/32[gre]
### swanctl -L
libgcrypt version mismatch
plugin 'gcrypt': failed to load - gcrypt_plugin_create returned NULL
ipsec_a2bc7f61-1: IKEv2, reauthentication every 10260s, no rekeying
  local:  0.0.0.0
  remote: xxx.xxx.0.167
  local public key authentication:
    id: bunker_ipsec
    certs: C=US, ST=CA, O=Open vSwitch, OU=Open vSwitch certifier, CN=bunker_ipsec
  remote public key authentication:
    id: bunker_ipsec
  ipsec_a2bc7f61-1: TRANSPORT, rekeying every 3060s
    local:  dynamic[gre]
    remote: dynamic[gre]
ipsec_9f44b2f4-1: IKEv2, reauthentication every 10260s, no rekeying
  local:  0.0.0.0
  remote: xxx.xxx.99.42
  local public key authentication:
    id: bunker_ipsec
    certs: C=US, ST=CA, O=Open vSwitch, OU=Open vSwitch certifier, CN=bunker_ipsec
  remote public key authentication:
    id: bunker_ipsec
  ipsec_9f44b2f4-1: TRANSPORT, rekeying every 3060s
    local:  dynamic[gre]
    remote: dynamic[gre]
### swanctl -l 
libgcrypt version mismatch
plugin 'gcrypt': failed to load - gcrypt_plugin_create returned NULL
ipsec_9f44b2f4-1: #2939, ESTABLISHED, IKEv2, 360f56b7865f874a_i* 8b4b4882937ab8e2_r
  local  'bunker_ipsec' @ xxx.xxx.99.160[4500]
  remote 'bunker_ipsec' @ xxx.xxx.99.42[4500]
  AES_GCM_16-256/PRF_HMAC_SHA2_256/MODP_2048
  established 1381s ago, reauth in 8172s
  ipsec_9f44b2f4-1: #112, reqid 4, INSTALLED, TRANSPORT, ESP:AES_GCM_16-256
    installed 1381s ago, rekeying in 1206s, expires in 2219s
    in  cc3b029c, 140760 bytes,  1380 packets,     1s ago
    out c1254456, 140760 bytes,  1380 packets,    44s ago
    local  xxx.xxx.99.160/32[gre]
    remote xxx.xxx.99.42/32[gre]
### swanctl --monitor-sa 
ipsec_9f44b2f4-1: #2897, DELETING, IKEv2, 2dcfba40b1927bbe_i* fb7f8620d5dc3df2_r
  local  'bunker_ipsec' @ xxx.xxx.99.160[4500]
  remote 'bunker_ipsec' @ xxx.xxx.99.42[4500]
  AES_GCM_16-256/PRF_HMAC_SHA2_256/MODP_2048
  passive: IKE_DELETE
  ipsec_9f44b2f4-1: #108, reqid 4, INSTALLED, TRANSPORT, ESP:AES_GCM_16-256
    installed 2008s ago, rekeying in 552s, expires in 1592s
    in  c4e7cfaa,   1632 bytes,    16 packets,     1s ago
    out cc55978a,   1632 bytes,    16 packets,     4s ago
    local  xxx.xxx.99.160/32[gre]
    remote xxx.xxx.99.42/32[gre]
ipsec_a2bc7f61-1: #2935, ESTABLISHED, IKEv2, 64a20c47dc46550b_i* d3da734c78c2df3f_r
  local  'bunker_ipsec' @ xxx.xxx.0.142[4500]
  remote 'bunker_ipsec' @ xxx.xxx.0.167[4500]
  AES_GCM_16-256/PRF_HMAC_SHA2_256/MODP_2048
  established 0s ago, reauth in 9947s
  active:  CHILD_CREATE IKE_AUTH_LIFETIME IKE_MOBIKE
  ipsec_a2bc7f61-1: #109, reqid 3, INSTALLED, TRANSPORT, ESP:AES_GCM_16-256
    installed 0s ago, rekeying in 2554s, expires in 3600s
    in  c8574ee8,      0 bytes,     0 packets
    out cae90843,      0 bytes,     0 packets
    local  xxx.xxx.0.142/32[gre]
    remote xxx.xxx.0.167/32[gre]
ipsec_a2bc7f61-1: #2935, DELETING, IKEv2, 64a20c47dc46550b_i* d3da734c78c2df3f_r
  local  'bunker_ipsec' @ xxx.xxx.0.142[4500]
  remote 'bunker_ipsec' @ xxx.xxx.0.167[4500]
  AES_GCM_16-256/PRF_HMAC_SHA2_256/MODP_2048
  passive: IKE_DELETE
  ipsec_a2bc7f61-1: #109, reqid 3, INSTALLED, TRANSPORT, ESP:AES_GCM_16-256
    installed 0s ago, rekeying in 2554s, expires in 3600s
    in  c8574ee8,      0 bytes,     0 packets
    out cae90843,      0 bytes,     0 packets
    local  xxx.xxx.0.142/32[gre]
    remote xxx.xxx.0.167/32[gre]
ipsec_9f44b2f4-1: #2936, ESTABLISHED, IKEv2, 09e8c5c4414c6d5c_i* 6066d7eaa3f18173_r
  local  'bunker_ipsec' @ xxx.xxx.99.160[4500]
  remote 'bunker_ipsec' @ xxx.xxx.99.42[4500]
  AES_GCM_16-256/PRF_HMAC_SHA2_256/MODP_2048
  established 0s ago, reauth in 10241s
  active:  CHILD_CREATE IKE_AUTH_LIFETIME IKE_MOBIKE
  ipsec_9f44b2f4-1: #110, reqid 4, INSTALLED, TRANSPORT, ESP:AES_GCM_16-256
    installed 0s ago, rekeying in 2530s, expires in 3600s
    in  c6ab35d4,      0 bytes,     0 packets
    out c67e0be1,      0 bytes,     0 packets
    local  xxx.xxx.99.160/32[gre]
    remote xxx.xxx.99.42/32[gre]
### swanctl --log
16[KNL] creating acquire job for policy xxx.xxx.0.142/32[gre] === xxx.xxx.0.167/32[gre] with reqid {3}
16[IKE] initiating IKE_SA ipsec_a2bc7f61-1[2938] to xxx.xxx.0.167
16[ENC] generating IKE_SA_INIT request 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) N(REDIR_SUP) ]
16[NET] sending packet: from xxx.xxx.0.142[500] to xxx.xxx.0.167[500] (1308 bytes)
11[NET] received packet: from xxx.xxx.0.167[500] to xxx.xxx.0.142[500] (489 bytes)
11[ENC] parsed IKE_SA_INIT response 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) CERTREQ N(FRAG_SUP) N(HASH_ALG) N(CHDLESS_SUP) N(MULT_AUTH) ]
11[CFG] selected proposal: IKE:AES_GCM_16_256/PRF_HMAC_SHA2_256/MODP_2048
11[IKE] received cert request for "C=US, ST=CA, O=Open vSwitch, OU=switchca, CN=OVS switchca CA Certificate (2023 Aug 15 09:59:54)"
11[IKE] sending cert request for "C=US, ST=CA, O=Open vSwitch, OU=switchca, CN=OVS switchca CA Certificate (2023 Aug 15 09:59:54)"
11[IKE] authentication of 'bunker_ipsec' (myself) with RSA_EMSA_PKCS1_SHA2_256 successful
11[IKE] sending end entity cert "C=US, ST=CA, O=Open vSwitch, OU=Open vSwitch certifier, CN=bunker_ipsec"
11[IKE] establishing CHILD_SA ipsec_a2bc7f61-1{111} reqid 3
11[ENC] generating IKE_AUTH request 1 [ IDi CERT CERTREQ IDr AUTH N(USE_TRANSP) SA TSi TSr N(MOBIKE_SUP) N(ADD_4_ADDR) N(ADD_4_ADDR) N(ADD_4_ADDR) N(MULT_AUTH) N(EAP_ONLY) N(MSG_ID_SYN_SUP) ]
11[ENC] splitting IKE message (1607 bytes) into 2 fragments
11[ENC] generating IKE_AUTH request 1 [ EF(1/2) ]
11[ENC] generating IKE_AUTH request 1 [ EF(2/2) ]
11[NET] sending packet: from xxx.xxx.0.142[4500] to xxx.xxx.0.167[4500] (1248 bytes)
11[NET] sending packet: from xxx.xxx.0.142[4500] to xxx.xxx.0.167[4500] (424 bytes)
08[NET] received packet: from xxx.xxx.99.42[4500] to xxx.xxx.99.160[4500] (65 bytes)
08[ENC] parsed INFORMATIONAL request 0 [ D ]
08[IKE] received DELETE for IKE_SA ipsec_9f44b2f4-1[2936]
08[IKE] deleting IKE_SA ipsec_9f44b2f4-1[2936] between xxx.xxx.99.160[bunker_ipsec]...xxx.xxx.99.42[bunker_ipsec]
08[IKE] IKE_SA deleted
08[ENC] generating INFORMATIONAL response 0 [ ]
08[NET] sending packet: from xxx.xxx.99.160[4500] to xxx.xxx.99.42[4500] (57 bytes)
06[NET] received packet: from xxx.xxx.0.167[4500] to xxx.xxx.0.142[4500] (1248 bytes)
06[ENC] parsed IKE_AUTH response 1 [ EF(1/2) ]
06[ENC] received fragment #1 of 2, waiting for complete IKE message
10[NET] received packet: from xxx.xxx.0.167[4500] to xxx.xxx.0.142[4500] (239 bytes)
10[ENC] parsed IKE_AUTH response 1 [ EF(2/2) ]
10[ENC] received fragment #2 of 2, reassembled fragmented IKE message (1422 bytes)
10[ENC] parsed IKE_AUTH response 1 [ IDr CERT AUTH N(USE_TRANSP) SA TSi TSr N(AUTH_LFT) N(MOBIKE_SUP) N(ADD_4_ADDR) N(ADD_4_ADDR) N(ADD_4_ADDR) ]
10[IKE] received end entity cert "C=US, ST=CA, O=Open vSwitch, OU=Open vSwitch certifier, CN=bunker_ipsec"
10[CFG]   using trusted ca certificate "C=US, ST=CA, O=Open vSwitch, OU=switchca, CN=OVS switchca CA Certificate (2023 Aug 15 09:59:54)"
10[CFG] checking certificate status of "C=US, ST=CA, O=Open vSwitch, OU=Open vSwitch certifier, CN=bunker_ipsec"
10[CFG] certificate status is not available
10[CFG]   reached self-signed root ca with a path length of 0
10[CFG]   using trusted certificate "C=US, ST=CA, O=Open vSwitch, OU=Open vSwitch certifier, CN=bunker_ipsec"
10[IKE] authentication of 'bunker_ipsec' with RSA_EMSA_PKCS1_SHA2_256 successful
10[IKE] IKE_SA ipsec_a2bc7f61-1[2938] established between xxx.xxx.0.142[bunker_ipsec]...xxx.xxx.0.167[bunker_ipsec]
10[IKE] scheduling reauthentication in 9991s
10[IKE] maximum IKE_SA lifetime 10531s
10[CFG] selected proposal: ESP:AES_GCM_16_256/NO_EXT_SEQ
10[IKE] CHILD_SA ipsec_a2bc7f61-1{111} established with SPIs cccaeead_i c06333c5_o and TS xxx.xxx.0.142/32[gre] === xxx.xxx.0.167/32[gre]
10[IKE] received AUTH_LIFETIME of 10027s, scheduling reauthentication in 9487s
10[IKE] peer supports MOBIKE
06[KNL] creating acquire job for policy xxx.xxx.99.160/32[gre] === xxx.xxx.99.42/32[gre] with reqid {4}
07[IKE] initiating IKE_SA ipsec_9f44b2f4-1[2939] to xxx.xxx.99.42
07[ENC] generating IKE_SA_INIT request 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) N(REDIR_SUP) ]
07[NET] sending packet: from xxx.xxx.99.160[500] to xxx.xxx.99.42[500] (1308 bytes)
13[NET] received packet: from xxx.xxx.99.42[500] to xxx.xxx.99.160[500] (489 bytes)
13[ENC] parsed IKE_SA_INIT response 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) CERTREQ N(FRAG_SUP) N(HASH_ALG) N(CHDLESS_SUP) N(MULT_AUTH) ]
13[CFG] selected proposal: IKE:AES_GCM_16_256/PRF_HMAC_SHA2_256/MODP_2048
13[IKE] received cert request for "C=US, ST=CA, O=Open vSwitch, OU=switchca, CN=OVS switchca CA Certificate (2023 Aug 15 09:59:54)"
13[IKE] sending cert request for "C=US, ST=CA, O=Open vSwitch, OU=switchca, CN=OVS switchca CA Certificate (2023 Aug 15 09:59:54)"
13[IKE] authentication of 'bunker_ipsec' (myself) with RSA_EMSA_PKCS1_SHA2_256 successful
13[IKE] sending end entity cert "C=US, ST=CA, O=Open vSwitch, OU=Open vSwitch certifier, CN=bunker_ipsec"
13[IKE] establishing CHILD_SA ipsec_9f44b2f4-1{112} reqid 4
13[ENC] generating IKE_AUTH request 1 [ IDi CERT CERTREQ IDr AUTH N(USE_TRANSP) SA TSi TSr N(MOBIKE_SUP) N(ADD_4_ADDR) N(ADD_4_ADDR) N(ADD_4_ADDR) N(MULT_AUTH) N(EAP_ONLY) N(MSG_ID_SYN_SUP) ]
13[ENC] splitting IKE message (1607 bytes) into 2 fragments
13[ENC] generating IKE_AUTH request 1 [ EF(1/2) ]
13[ENC] generating IKE_AUTH request 1 [ EF(2/2) ]
13[NET] sending packet: from xxx.xxx.99.160[4500] to xxx.xxx.99.42[4500] (1248 bytes)
13[NET] sending packet: from xxx.xxx.99.160[4500] to xxx.xxx.99.42[4500] (424 bytes)
07[NET] received packet: from xxx.xxx.0.167[4500] to xxx.xxx.0.142[4500] (65 bytes)
07[ENC] parsed INFORMATIONAL request 0 [ D ]
07[IKE] received DELETE for IKE_SA ipsec_a2bc7f61-1[2938]
07[IKE] deleting IKE_SA ipsec_a2bc7f61-1[2938] between xxx.xxx.0.142[bunker_ipsec]...xxx.xxx.0.167[bunker_ipsec]
07[IKE] IKE_SA deleted
07[ENC] generating INFORMATIONAL response 0 [ ]
07[NET] sending packet: from xxx.xxx.0.142[4500] to xxx.xxx.0.167[4500] (57 bytes)
14[NET] received packet: from xxx.xxx.99.42[4500] to xxx.xxx.99.160[4500] (1248 bytes)
14[ENC] parsed IKE_AUTH response 1 [ EF(1/2) ]
14[ENC] received fragment #1 of 2, waiting for complete IKE message
12[NET] received packet: from xxx.xxx.99.42[4500] to xxx.xxx.99.160[4500] (239 bytes)
12[ENC] parsed IKE_AUTH response 1 [ EF(2/2) ]
12[ENC] received fragment #2 of 2, reassembled fragmented IKE message (1422 bytes)
12[ENC] parsed IKE_AUTH response 1 [ IDr CERT AUTH N(USE_TRANSP) SA TSi TSr N(AUTH_LFT) N(MOBIKE_SUP) N(ADD_4_ADDR) N(ADD_4_ADDR) N(ADD_4_ADDR) ]
12[IKE] received end entity cert "C=US, ST=CA, O=Open vSwitch, OU=Open vSwitch certifier, CN=bunker_ipsec"
12[CFG]   using trusted ca certificate "C=US, ST=CA, O=Open vSwitch, OU=switchca, CN=OVS switchca CA Certificate (2023 Aug 15 09:59:54)"
12[CFG] checking certificate status of "C=US, ST=CA, O=Open vSwitch, OU=Open vSwitch certifier, CN=bunker_ipsec"
12[CFG] certificate status is not available
12[CFG]   reached self-signed root ca with a path length of 0
12[CFG]   using trusted certificate "C=US, ST=CA, O=Open vSwitch, OU=Open vSwitch certifier, CN=bunker_ipsec"
12[IKE] authentication of 'bunker_ipsec' with RSA_EMSA_PKCS1_SHA2_256 successful
12[IKE] IKE_SA ipsec_9f44b2f4-1[2939] established between xxx.xxx.99.160[bunker_ipsec]...xxx.xxx.99.42[bunker_ipsec]
12[IKE] scheduling reauthentication in 9925s
12[IKE] maximum IKE_SA lifetime 10465s
12[CFG] selected proposal: ESP:AES_GCM_16_256/NO_EXT_SEQ
12[IKE] CHILD_SA ipsec_9f44b2f4-1{112} established with SPIs cc3b029c_i c1254456_o and TS xxx.xxx.99.160/32[gre] === xxx.xxx.99.42/32[gre]
12[IKE] received AUTH_LIFETIME of 10093s, scheduling reauthentication in 9553s
12[IKE] peer supports MOBIKE
10[NET] received packet: from xxx.xxx.0.167[500] to xxx.xxx.0.142[500] (1308 bytes)
10[ENC] parsed IKE_SA_INIT request 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) N(REDIR_SUP) ]
10[IKE] received retransmit of request with ID 0, retransmitting response
10[NET] sending packet: from xxx.xxx.0.142[500] to xxx.xxx.0.167[500] (489 bytes)

#### Abnormal packet capture results
![20230828-112231(WeLinkPC)](https://github.com/strongswan/strongswan/assets/53250153/666329b5-21b2-4d49-8e19-4316a35fce73)
#### Normal packet capture results
![20230828-112243(WeLinkPC)](https://github.com/strongswan/strongswan/assets/53250153/ecd67493-2a2a-41fe-aa63-4379458d83db)

**Comment by tobiasbrunner** [comment]
Using the same identities for two connections to the same host that you want to establish concurrently obviously won't work if `unqiueids=yes` is configured. Also, you don't want the _duplicheck_ plugin active, it doesn't do what you might think it does.
